include/posix/pthread.h

```diff
@@ -383,4 +393,151 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Base Pthread related APIs */
+
+/**
+ * @brief get pid of current execueting thread.
```

Comment:
pthread_self() is not necessarily a PID; in fact, Zephyr doesn't have the notion of process ID.  `man 3p pthread_self` here says "get the calling thread ID", which I think is more appropriate.

Reply:
Addressed